plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1908,20 +1909,46 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
                         "%s cannot be downgraded from %d to %d", WRITER_VERSION_PROPERTY, currentProtocolEntry.getMinWriterVersion(), writerVersion.get()));
             }
         }
+        long createdTime = Instant.now().toEpochMilli();
+
+        Map<String, String> existingConfiguration = handle.getMetadataEntry().getConfiguration();
+
+        Optional<MetadataEntry> metadataEntry = Optional.empty();
+        if (properties.containsKey(CHANGE_DATA_FEED_ENABLED_PROPERTY)) {
+            boolean changeDataFeedEnabled = (Boolean) properties.get(CHANGE_DATA_FEED_ENABLED_PROPERTY)
+                    .orElseThrow(() -> new IllegalArgumentException("The change_data_feed_enabled  property cannot be empty"));
+            Map<String, String> configuration = new HashMap<>(existingConfiguration);
+            configuration.put(DELTA_CHANGE_DATA_FEED_ENABLED_PROPERTY, String.valueOf(changeDataFeedEnabled));
+            metadataEntry = Optional.of(buildMetadataEntry(session, handle, configuration, createdTime));
+            long upToDateWriterVersion = writerVersion.orElse(currentProtocolEntry.getMinWriterVersion());
+            if (changeDataFeedEnabled && (upToDateWriterVersion < CDF_SUPPORTED_WRITER_VERSION)) {
```

Comment:
nit: I would move this check to just after `boolean changeDataFeedEnabled`. There's no need to build `MetadataEntry` object in case of invalid table properties.